api/src/multiaccessy/accessy.py

```diff
@@ -42,107 +46,215 @@ def create_session(cls, client_id: str, client_secret: str) -> "AccessySession":
+        return self._get_users_in_access_group(ACCESSY_LAB_ACCESS_GROUP)
+
+    def _get_users_special(self) -> list[dict]:
+        """ Get all user ID:s with special access """
+        return self._get_users_in_access_group(ACCESSY_SPECIAL_ACCESS_GROUP)
+    
+    def _user_ids_to_accessy_members(self, user_ids: list[UUID]) -> list["AccessyMember"]:
+        """ Convert a list of User ID:s to AccessyMembers """
+
+        APPLICATION_PHONE_NUMBER = object()  # Sentinel phone number for applications
+        def fill_user_details(user: "AccessyMember", user_id: str):
+            data = self._get_json(f"/org/admin/user/{user_id}")
+
+            # API keys do not have phone numbers
+            if data["application"]:
+                user.phone_number = APPLICATION_PHONE_NUMBER
+                return
+
+            try:
+                user.phone_number = data["msisdn"]
+            except KeyError:
+                logger.error(f"User {user.user_id} does not have a phone number. {data=}")
+            user.first_name = data.get("firstName", "")
+            user.last_name = data.get("lastName", "")
+
+        def fill_membership_id(user: "AccessyMember", user_id: str):
+            data = self._get_json(f"/asset/admin/user/{user_id}/organization/{self.organization_id}/membership")
+            user.membership_id = data["id"]
+        
+        threads = []
+        accessy_members = []
+        for uid in user_ids:
+            accessy_member = AccessyMember(uid, None)
+            threads.append(threading.Thread(target=fill_user_details, args=(accessy_member, uid)))
+            threads.append(threading.Thread(target=fill_membership_id, args=(accessy_member, uid)))
+            accessy_members.append(accessy_member)
+        
+        for t in threads:
+            t.start()
 
-    def get_membership(self, user_id: str) -> str:
-        response = requests.get(ACCESSY_URL + f"/asset/admin/user/{user_id}/organization/{self.organization_id}/membership",
-                                headers={"Authorization": f"Bearer {self.session_token}"})
+        for t in threads:
+            t.join()
+        
+        # Filter out API keys
+        accessy_members = [m for m in accessy_members if m.phone_number is not APPLICATION_PHONE_NUMBER]
 
-        check_response_error(response, "Get membership")
-        data = response.json()
-        return data["id"]
+        return accessy_members
+    
+    def _get_org_user_from_phone(self, phone_number: MSISDN, users_in_org: list[dict] = None) -> Union[None, "AccessyMember"]:
+        """ Get a AccessyMember from a phone number (if in org) """
+        if users_in_org is None:
+            users_in_org = self._get_users_org()
+
+        for item in users_in_org:
+            if item.get("msisdn", None) == phone_number:
+                user_id = item["id"]
+                return self._user_ids_to_accessy_members([user_id])[0]
+        else:
+            return None
+    
+    #################
```

Comment:
Maybe easier to read if public methods are at the top.

Reply:
Moved public to top